### PR TITLE
promote i586-unknown-linux-gnu to Tier2+hosttools

### PR DESCRIPTION
**Pull request by goshhhy**
mcp: https://github.com/rust-lang/compiler-team/issues/543

### CODE REVIEW

File: src/ci/github-actions/ci.yml [hunk 1]
```diff
@@ -347,7 +347,10 @@ jobs:
           - name: dist-armv7-linux
             <<: *job-linux-xl
 
-          - name: dist-i586-gnu-i586-i686-musl
+          - name: dist-i586-i686-musl
+            <<: *job-linux-xl
+
+          - name: dist-i586-linux
```
Comment (workingjubilee) [inline src/ci/github-actions/ci.yml:353]: Note:
https://github.com/rust-lang/rust/blob/1266e8d25753d89af1e06667b00e87944350f332/src/ci/github-actions/ci.yml#L1-L31

**Comment by workingjubilee** [comment]
> [293/3023] Linking CXX executable bin/llvm-tblgen
FAILED: bin/llvm-tblgen 
 (snip a bunch of hot nonsense)
/usr/bin/ld: cannot find /lib/libpthread.so.0: No such file or directory
/usr/bin/ld: cannot find /usr/lib/libpthread_nonshared.a: No such file or directory
collect2: error: ld returned 1 exit status
 (snip building a few other items)
ninja: build stopped: subcommand failed.

Looks like we need to throw more stuff where LLVM can find it.

**Comment by Dylan-DPC** [comment]
@goshhhy any updates on this?

**Comment by workingjubilee** [comment]
Oh, this is probably stuck without further guidance. I was going to investigate and try to help but I wound up distracted, unfortunately.

**Comment by JohnCSimon** [comment]
@goshhhy 
Ping from triage: I'm closing this due to inactivity, Please reopen when you are ready to continue with this. 
Note: if you are going to continue please open the PR **BEFORE** you push to it, else you won't be able to reopen - this is a quirk of github. 
Thanks for your contribution.

@rustbot label: +S-inactive

**Pull request closed without merging**